src/Provider/IslandoraServiceProvider.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Islandora\Crayfish\Commons\Provider;
+
+use Islandora\Crayfish\Commons\CmdExecuteService;
+use Islandora\Crayfish\Commons\FedoraResourceConverter;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Silex\Provider\DoctrineServiceProvider;
+use Silex\Provider\MonologServiceProvider;
+use Silex\Provider\ServiceControllerServiceProvider;
+use Silex\Provider\SecurityServiceProvider;
+use Islandora\Chullo\FedoraApi;
+use Islandora\Crayfish\Commons\Syn\SettingsParser;
+use Islandora\Crayfish\Commons\Syn\JwtAuthenticator;
+use Islandora\Crayfish\Commons\Syn\JwtFactory;
+
+class IslandoraServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register(Container $container)
+    {
+        // Register services we rely on
+        $container->register(new MonologServiceProvider());
+        $container->register(new ServiceControllerServiceProvider());
+        $container->register(new SecurityServiceProvider());
+        $container->register(new DoctrineServiceProvider());
+
+        // Configure external services
+        $container['monolog.logfile'] = function ($container) {
+            return strtolower($container['crayfish.log.level']) == 'none' ? null : $container['crayfish.log.file'];
+        };
+        $container['monolog.level'] = function ($container) {
+            return $container['crayfish.log.level'];
```

Comment:
Ok so it seemed that the issue was that `IslandoraServiceProvider` is trying to use the `crayfish.log.level` when it is instantiated. Which is before the YamlConfigServiceProvider is registered and so there are no configuration options? But that seems like it would never have worked for you. 

However, if I switch the order then I get errors around the missing definiton of `$app['db']`.

I'll try cleaning this out and start again.